python/mxnet/test_utils.py

```diff
@@ -89,23 +89,34 @@ def same(a, b):
     """
     return np.array_equal(a, b)
 
-def reldiff(a, b):
+def reldiff(a, b, return_index=False, thresh=1E-3):
     """Calculate the relative difference between two input arrays
 
-    Calculated by :math:`\\frac{|a-b|_1}{|a|_1 + |b|_1}`
+    Calculated by :math:`\\max(frac{|a-b|}{max(|a|, |b|) + 1E-7})`
+    Also If |a-b| is smaller than "thresh", the absolute difference will be used.
 
     Parameters
     ----------
     a : np.ndarray
     b : np.ndarray
+    return_index : bool, optional
+        Whether to return the index that has the maximum difference
+    thresh : float, optional
+        If |a-b| is smaller than "thresh", the absolute difference will be used.
     """
-    diff = np.sum(np.abs(a - b))
-    norm = np.sum(np.abs(a)) + np.sum(np.abs(b))
-    if diff == 0:
-        return 0
-    ret = diff / norm
-    return ret
-
+    diff = np.abs(a - b)
+    norm = np.maximum(np.abs(a), np.abs(b)) + 1e-7
+    relative_diff = diff / norm
+    if isinstance(diff, np.ndarray):
```

Comment:
assert_allclose uses a per entry atol + rtol*x threshold. I think its better than the hacky switch here.
We should do something similar. Changing them all shouldn't be too hard since it's usually just assert reldiff() < 0.01 anyway. What custom behavior do we need?